pkgs/dart_tooling_mcp_server/lib/src/mixins/dtd.dart

```diff
@@ -201,6 +211,89 @@ base mixin DartToolingDaemonSupport on ToolsSupport {
     );
   }
 
+  /// Retrieves runtime errors from the currently running app.
+  ///
+  /// If more than one debug session is active, then it just uses the first one.
+  ///
+  // TODO: support passing a debug session id when there is more than one debug
+  // session.
+  Future<CallToolResult> runtimeErrors(CallToolRequest request) async {
+    return _callOnVmService(
+      callback: (vmService) async {
+        final errors = <String>[];
+        try {
+          await vmService.streamListen(EventStreams.kExtension);
```

Comment:
Not sure if these are broadcast streams or not - but if they are (or we want to be defensive) then we should start listening after setting up the stream listeners below.